src/http.cpp

```diff
@@ -546,13 +556,11 @@ Request::build()
         request << "Connection: " << conn_str << "\r\n";
 
     // body & content-length
-    if (!body_.empty()){
+    if (!body_.empty())
         request << "Content-Length: " << body_.size() << "\r\n\r\n";
-        request << body_;
-    }
-
     // last delim
-    request << "\r\n";
+    if (append_body)
```

Comment:
body_ should be non empty.

Reply:
@AmarOk1412 in a case of 100-continue yes! its the purpose of such optimization is send the header only before continuing with a body, thats why I set the variable append_body in such a header is present! line 538: ```if (header.first == restinio::http_field_t::expect and header.second == "100-continue")```

Reply:
I know. What I mean is to add the append_body only if body_.empty() is also false; Like

if (append_body && !body.empty())

Reply:
but then, I will have to trim the raw request that will be written on a socket during a send(), thus, I just build() one without a body and send the body separately after in the sequence if such option is set!